tests/publiccloud/prepare_tools.pm

```diff
@@ -122,6 +122,30 @@ EOT
     validate_script_output("terraform -version", qr/$terraform_version/);
     record_info('Terraform', script_output('terraform -version'));
 
+    # Ansible install with pip
+    # Default version is chosen as low as possible so it run also on SLE12's
+    # ANSIBLE_CORE_VERSION should be set only if the different then default one need to be used
+    my $ansible_version = get_var('ANSIBLE_VERSION', '4.10.0');
+    my $ansible_core_version = get_var('ANSIBLE_CORE_VERSION');
+    my $ansible_install_log = '/tmp/ansible_install.log';
+
+    if ( script_run("python3.9 -m pip -V") == 0 ) {
+        assert_script_run("python3.9 -m pip install --no-input -q --no-color --log $ansible_install_log ansible==$ansible_version", timeout => 240 );
+        upload_logs( "$ansible_install_log" , failok => 1 );
+    } else {
+        assert_script_run("curl https://bootstrap.pypa.io/get-pip.py -o /tmp/get-pip.py", timeout => 240 );
+        assert_script_run("python3.9 /tmp/get-pip.py", timeout => 240 );
+        assert_script_run("python3.9 -m pip install --no-input -q --no-color --log $ansible_install_log ansible==$ansible_version", timeout => 240 );
+        upload_logs( "$ansible_install_log" , failok => 1 );
+    }
+
+    if ( length $ansible_core_version ) {
+        my $ansible_core_install_log = "/tmp/ansible_core_install.log";
+        assert_script_run("python3.9 -m pip install --no-input -q --no-color --log $ansible_core_install_log ansible-core==$ansible_core_version", timeout => 240 );
+        upload_logs( "$ansible_core_install_log" , failok => 1 );
+    }
+    record_info('Ansible', script_output('ansible --version'));
```

Comment:
Good idea! This also serves as a smoketest to check if `ansible` is actually installed.